llvm/lib/Target/X86/lab3/kuznetsov_artyom/CMakeLists.txt

```diff
@@ -0,0 +1,10 @@
+set(PluginName X86MulAddPass)
+file(GLOB_RECURSE ALL_SOURCE_FILES *.cpp *.h)
+add_llvm_pass_plugin(${PluginName}
+        ${ALL_SOURCE_FILES}
+        DEPENDS
+        intrinsics_gen
+        X86
+        BUILDTREE_ONLY)
+
+target_include_directories(${PluginName} PUBLIC ${PATH_TO_X86})
```

Comment:
```suggestion
target_include_directories(${PluginName} PUBLIC ${PATH_TO_X86})

set(LLVM_TEST_DEPENDS ${PluginName} ${LLVM_TEST_DEPENDS} PARENT_SCOPE)

```

Reply:
P.S : There is no need to add this dependency, when building your plugin, it is enough to prescribe `cmake --build . --target NamePass`, where we specify `NamePass` in `CMakeLists.txt` . And in the pool revquest, the project builds completely, so our targets are automatically picked up

Reply:
I've double-checked it now, even if I write `cmake --build . --target check-llvm-lab3` on my system, then the changes made to the file `X86MulAddPass.cpp` they don't catch on. The same will happen with `cmake --build . --target check-llvm`